### PR TITLE
Fix various issues with new ImageBuilder pipeline step templates

### PR DESCRIPTION
**Pull request by lbussell**
Fixes various issues from https://github.com/dotnet/dotnet-docker/pull/5421.

- Use authentication for Build command
- Use PowerShell env var to reference dryRunArg variable
- Use Windows PowerShell for Windows jobs

There is still one outstanding issue - our Windows agents don't have Az CLI installed. That is currently being addressed at the VM image level but I am working on a workaround now. This PR will fail internal Windows builds until that is addressed so this is in draft mode.

### CODE REVIEW

File: eng/common/templates/steps/run-imagebuilder.yml [hunk 1]
```diff
@@ -14,23 +17,31 @@ parameters:
...
     displayName: ${{ parameters.displayName }}
     continueOnError: ${{ parameters.continueOnError }}
     inputs:
       azureSubscription: ${{ parameters.serviceConnection }}
       addSpnToEnvironment: true
-      scriptType: 'pscore'
+      ${{ if eq(parameters.dockerClientOS, 'linux') }}:
+        scriptType: 'ps'
+      ${{ else }}:
+        scriptType: 'pscore'
```
[the remark this answers is not in the document]
Comment (lbussell) [reply eng/common/templates/steps/run-imagebuilder.yml:36]: Yes. I forgot to update this branch with my latest changes from internal. Will update shortly.

**Comment by lbussell** [comment]
Windows VM Images are updated to contain Az CLI and I fixed some additional bugs, everything seems to be working now! [Test run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2444735&view=results) [internal link]

**Review by mthalman — APPROVED** [review]
Have you also ran a test for the public build to check that things are fixed there as well?

**Comment by mthalman** [comment]
> @mthalman Appears to be working: [dotnet/dotnet-docker#5437](https://github.com/dotnet/dotnet-docker/pull/5437)
> 
> This pull request's pipeline runs are also evidence of that.

That one has an error in CopyBaseImages though: https://dev.azure.com/dnceng-public/public/_build/results?buildId=666898&view=logs&jobId=5e8a216f-208e-5f25-daac-ba5cedb89a59&j=1a3858d4-df16-5910-9f2f-b79bed34300b&t=cbac5fd7-97ef-5b4a-4622-7378ec20d382

**Comment by lbussell** [comment]
> > @mthalman Appears to be working: [dotnet/dotnet-docker#5437](https://github.com/dotnet/dotnet-docker/pull/5437)
> > This pull request's pipeline runs are also evidence of that.
> 
> That one has an error in CopyBaseImages though: https://dev.azure.com/dnceng-public/public/_build/results?buildId=666898&view=logs&jobId=5e8a216f-208e-5f25-daac-ba5cedb89a59&j=1a3858d4-df16-5910-9f2f-b79bed34300b&t=cbac5fd7-97ef-5b4a-4622-7378ec20d382

Oops, it had some outdated changes. It's running again now.

**Comment by mthalman** [comment]
But you're right, this PR's build is essentially doing the same thing and it's running clean.

**Comment by lbussell** [comment]
Successful public run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=666933&view=results

**Pull request merged**